components/eventing-controller/api/v1alpha1/subscription_conversion.go

```diff
@@ -1,19 +1,340 @@
 package v1alpha1
 
 import (
+	"fmt"
+	"strconv"
+	"strings"
+
 	"github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha2"
+	"github.com/pkg/errors"
 	"sigs.k8s.io/controller-runtime/pkg/conversion"
 )
 
+const (
+	hubVersionErrMsg     = "hub version is not the expected v1alpha2 version"
+	multipleSourceErrMsg = "subscription contains more than 1 eventSource"
+)
+
 // ConvertTo converts this Subscription to the Hub version (v2).
-func (s *Subscription) ConvertTo(rawObj conversion.Hub) error {
-	_ = rawObj.(*v1alpha2.Subscription)
+func (src *Subscription) ConvertTo(dstRaw conversion.Hub) error { // nolint
```

Comment:
Why do we need the `nolint` ? If we really need it, there should be a comment explaining why we need to exclude linting IMHO

Reply:
Same question for the other occurences :)

Reply:
The problem is the naming of the Subscription, on which we call this method. In the ConvertTo its' `src` and in the ConvertFrom it's the `dst`, which makes sense. But linter doesn't seem to like it.

Reply:
```
$ revive api/v1alpha1/subscription_conversion.go
api/v1alpha1/subscription_conversion.go:77:1: receiver name dst should be consistent with previous receiver name src for Subscription
```